app/entities/query.go

```diff
@@ -16,22 +16,28 @@ type QueryParams struct {
 
 // Validate checks if QueryParameter fields are valid.
 func (qp QueryParams) Validate() error {
+	if qp.Limit == "" {
+		return fmt.Errorf("limit argument has to be set")
+	}
+
 	if _, err := strconv.Atoi(qp.Limit); err != nil {
 		return fmt.Errorf("limit argument has to be a number")
 	}
 
+	if qp.Offset == "" {
+		return fmt.Errorf("offset argument has to be set")
+	}
+
 	if _, err := strconv.Atoi(qp.Offset); err != nil {
 		return fmt.Errorf("offset argument has to be a number")
 	}
 
-	allowedSortArgs := []string{"first_name", "last_name", "created_at", ","}
-
-	for _, arg := range allowedSortArgs {
-		qp.Sort = strings.ReplaceAll(qp.Sort, arg, "")
-	}
+	allowedSortArgs := map[string]bool{"first_name": true, "last_name": true, "created_at": true}
```

Comment:
If it is a Set (data structure with values that don't duplicate) - use `map[string]struct{}` (empty struct a) weights 0 bytes b) obvious that there won't be both true and false entries there).

Reply:
I have already tried usign {} and it looks ugly when I print it out.

Reply:
I often use slices for that things, to be honest.

Reply:
O^2 complexity is not acceptable 😄